### PR TITLE
Give proper errors for incorrect application of lazy builtins

### LINKED ISSUE #1828: Give proper errors for incorrect application of lazy builtins
Currently, the Internal to Core translation crashes if a lazy builtin (if, ||, etc) is applied to a wrong number of arguments. This should be checked during arity checking or type checking, and a user-level JuvixError should be thrown.

### PR DESCRIPTION
**Pull request by lukaszcz**
* Closes #1828

### CODE REVIEW

**Review by janmasrovira — DISMISSED** [review]

File: src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs [hunk 1]
```diff
@@ -488,8 +488,74 @@ checkExpression hintArity expr = case expr of
...
+                (_ : _ : _) -> appHelper f args
+                _ ->
+                  throw $
+                    ErrBuiltinNotFullyApplied
+                      BuiltinNotFullyApplied
+                        { _builtinNotFullyAppliedName = n,
+                          _builtinNotFullyAplliedExpectedArgsNum = 2
+                        }
+            _ -> appHelper f args
+        _ ->
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs:557]: Okay, I'll do that

[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Checker.hs]: Done

File: src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Error.hs [hunk 2]
```diff
@@ -15,6 +15,7 @@ data ArityCheckerError
   | ErrPatternFunction PatternFunction
   | ErrTooManyArguments TooManyArguments
   | ErrFunctionApplied FunctionApplied
+  | ErrBuiltinNotFullyApplied BuiltinNotFullyApplied
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Error.hs:18]: Done

File: src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Error/Types.hs [hunk 3]
```diff
@@ -232,3 +232,31 @@ instance ToGenericError FunctionApplied where
...
+        where
+          opts' = fromGenericOptions opts
+          i = getLoc (e ^. builtinNotFullyAppliedName)
+          argsNum = e ^. builtinNotFullyAplliedExpectedArgsNum
+          msg =
+            "The lazy builtin"
+              <+> ppCode opts' (e ^. builtinNotFullyAppliedName)
+              <+> "must be applied to exactly"
+              <+> pretty argsNum
+              <+> "arguments."
```
[the remark this answers is not in the document]
Comment (lukaszcz) [reply src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/ArityChecking/Error/Types.hs]: Done

**Review by janmasrovira — APPROVED** [review]

**Pull request merged**